main.go

```diff
@@ -74,13 +48,32 @@ func (hc *HealthCheckWebService) apiGetIsReplica(w http.ResponseWriter, r *http.
 		panic(err)
 	}
 
+	// if byte lag exceeds max_allowable_byte_lag then return 500
+	if hc.maxAllowableByteLagExceeded(r, nodeInfo) {
+		w.WriteHeader(http.StatusServiceUnavailable)
+	}
+
 	if !nodeInfo.IsReplica() {
 		w.WriteHeader(http.StatusServiceUnavailable)
 	}
 
 	json.NewEncoder(w).Encode(nodeInfo)
 }
 
+func maxAllowableByteLagExceeded(r *http.Request, nodeInfo *NodeInfo) bool {
```

Comment:
```
func (hc *HealthCheckWebService) maxAllowableByteLagExceeded(r *http.Request, nodeInfo *NodeInfo) bool {
...
```

With the `(hc *HealthCheckWebService)` after `func` it makes this a member function on the `HealthCheckWebService` struct, so you don't need to pass it as a variable on `HealthCheckWebService` from the `main` func.